### PR TITLE
Plugins:  propagate plugin exception

### LINKED ISSUE #8057: Plugins:  exception details lost during plugin creation
If plugin creation fails, the exception details are lost [here](https://github.com/NuGet/NuGet.Client/blob/bd0c5fb12076a7f1678d825541a978caa4ac0bc0/src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs#L255).

Then, a "A task was canceled." error is reported with no callstack.

### PR DESCRIPTION
**Pull request by dtivel**
Resolve https://github.com/NuGet/Home/issues/8057.

Ideally, [`NuGet.Common.ILogger`](https://github.com/NuGet/NuGet.Client/blob/5b975f446808a318e0696c8c23cd2dd6b9aa817b/src/NuGet.Core/NuGet.Common/Logging/ILogger.cs#L12) would have a method that would take an exception and log the callstack depending on the logging verbosity level:
```C#
    void LogException(Exception exception);
```
However, that change is a large, breaking change and out of scope for fixing https://github.com/NuGet/Home/issues/8057.

As it currently stands, it's left to the caller of an `ILogger` implementation to determine whether or not to log a callstack based on the logging verbosity level, but the logging verbosity level is unavailable to the caller.

The simple solution now is to log the callstack with `ILogger.LogDebug(…)` and leave it to the logger to decide whether or not to log debug events.

### CODE REVIEW

File: NuGet.sln [hunk 1]
```diff
@@ -624,7 +624,7 @@ Global
 		{50E33DA2-AF14-486D-81B8-BD8409744A38} = {08F523EC-3C2A-4A00-A54C-2E54C5AC856B}
 		{23725516-0A6E-48F9-A148-B5E3FA515B13} = {00CE0FBE-DA5A-4AD8-B0D6-04ECE6B04F0D}
 		{5039C2F9-CAD9-4162-849C-5EF7E3CE03D9} = {01BC4531-1E25-48D7-A8FD-A47D6FEC47FB}
-		{3DB4EDE3-B57B-498C-9EA9-A8BF8D9D4B2B} = {01BC4531-1E25-48D7-A8FD-A47D6FEC47FB}
+		{3DB4EDE3-B57B-498C-9EA9-A8BF8D9D4B2B} = {7323F93B-D141-4001-BB9E-7AF14031143E}
```
Comment (dtivel) [inline NuGet.sln:627]: Before this change NuGet.Credentials.Test showed up in Visual Studio under the NuGet.Client.UnitTests folder, even though the project is under test\NuGet.Core.Tests.  After this change, the project shows up under the NuGet.Core.UnitTests folder.

File: runTests.ps1 [hunk 2]
```diff
@@ -87,9 +87,9 @@ Invoke-BuildStep 'Cleaning package cache' {
     -skip:(-not $CI) `
     -ev +BuildErrors
 
-Invoke-BuildStep 'Running /t:RestoreVS15' {
+Invoke-BuildStep 'Running /t:RestoreVS' {
```
Comment (dtivel) [inline runTests.ps1:90]: These changes were overlooked by https://github.com/NuGet/NuGet.Client/pull/2832.

Comment (nkolev92) [reply runTests.ps1:90]: My bad, we don't use these scripts in our build at all, so I didn't even think about them.

**Review by nkolev92 — APPROVED** [review]
Looks great. 
👏

**Pull request merged**